dot/state/block_test.go

```diff
@@ -482,14 +479,7 @@ func TestAddBlockToBlockTree(t *testing.T) {
 }
 
 func TestNumberIsFinalised(t *testing.T) {
-	ctrl := gomock.NewController(t)
-
-	triesGauge := NewMockGauge(ctrl)
-	triesGauge.EXPECT().Set(0.00).Times(2)
-	tries := &Tries{
-		rootToTrie: make(map[common.Hash]*trie.Trie),
-		triesGauge: triesGauge,
-	}
```

Comment:
Can you add your gauges to the previous test code instead of removing it? It should only take ~4 more lines for the 2 new gauges. We should test our metrics (at least new ones we add).

Reply:
I don't see the need to test gauge mocks in every test that utilises the in-memory tries.  We should have a test to solely test metrics functionality.

Reply:
I added the mock counters to `trie_test.go`.

Reply:
> I don't see the need to test gauge mocks in every test that utilises the in-memory tries. We should have a test to solely test metrics functionality.

I think we should mock everything that is called as long as it's in the same package, to have better testing depth / more passes on the same code. Although it doesn't test the metrics functionality directly, you can find other inconsistencies with this strategy. But up to you, juts sharing my 2 cents 😉

Reply:
For each test that uses `Tries` for metric I would have to create a mock gauge or counter, and copy and paste the intended behaviour of the mock.  I think it's overkill.  IMO metric tests should be within the unit tests of the type that is publishing the metrics.

Reply:
> overkill

Agreed! But a bit like those stress tests that helped us find the flags errors :eyes: Not their direct purpose, but it can indirectly fish out some unexpected behavior. Anyway resolving this!